pygmt/helpers/utils.py

```diff
@@ -222,30 +220,72 @@ def data_kind(
 
     Examples
     --------
+    >>> import io
+    >>> from pathlib import Path
     >>> import numpy as np
+    >>> import pandas as pd
     >>> import xarray as xr
-    >>> import pathlib
-    >>> import io
-    >>> data_kind(data=None)
-    'vectors'
-    >>> data_kind(data=np.arange(10).reshape((5, 2)))
-    'matrix'
-    >>> data_kind(data="my-data-file.txt")
-    'file'
-    >>> data_kind(data=pathlib.Path("my-data-file.txt"))
-    'file'
+
+    The "arg" kind:
+
+    >>> [data_kind(data=data, required=False) for data in (2, 2.0, True, False)]
+    ['arg', 'arg', 'arg', 'arg']
     >>> data_kind(data=None, required=False)
     'arg'
-    >>> data_kind(data=2.0, required=False)
-    'arg'
-    >>> data_kind(data=True, required=False)
-    'arg'
-    >>> data_kind(data=xr.DataArray(np.random.rand(4, 3)))
+
+    The "file" kind:
+
+    >>> [data_kind(data=data) for data in ("file.txt", ("file1.txt", "file2.txt"))]
+    ['file', 'file']
+    >>> data_kind(data=Path("file.txt"))
+    'file'
+    >>> data_kind(data=(Path("file1.txt"), Path("file2.txt")))
+    'file'
+
+    The "grid" kind:
+
+    >>> data_kind(data=xr.DataArray(np.random.rand(4, 3)))  # 2-D xarray.DataArray
+    'grid'
+    >>> data_kind(data=xr.DataArray(np.arange(12)))  # 1-D xarray.DataArray
+    'grid'
+    >>> data_kind(data=xr.DataArray(np.random.rand(2, 3, 4, 5)))  # 4-D xarray.DataArray
     'grid'
```

Comment:
Erm, do we need tests to check that 1-D and 4-D xarray.DataArray are counted as grid?

Reply:
Not neccessary. These doctests are added to understand the current behavior of the `data_kind` function. 

I feel 1-D xarray.DataArray should be considered as a vector and 4-D xarray.DataArray should not be recognized and we should raise an exception. Will refactor the `data_kind` function in a separate PR.

Reply:
> I feel 1-D xarray.DataArray should be considered as a vector

We already recognize an `xr.Dataset` composed of 1-D `xarray.DataArray`s as a vector (can be treated as multiple 'columns' in a Dataframe). I don't think we need to expand this logic, since they users can call `.data` on a 1-D `xarray.DataArray` to get a numpy.ndarray that can then be passed to the `data` parameter.

> 4-D xarray.DataArray should not be recognized and we should raise an exception

Logic could be something like `if dataarray.ndim() >= 4: raise Error`, but I don't think this is really necessary since an error would be raised by the PyGMT module anyway right?